docs/examples/handling-complex-routes.md

```diff
@@ -0,0 +1,86 @@
+---
+description: |
+  Sometimes URL based routing isn't enough.
+---
+
+The page on [routing](/concepts/routing) hints at complex routing based on URL
+patterns using a `RouteConfig` object. Let's dive into this in a bit more
+detail.
+
+A `RouteConfig` has a `routeOverride` string property, which makes use of the
+[URL Pattern API](https://developer.mozilla.org/en-US/docs/Web/API/URL_Pattern_API).
+Here you can define named groups, wildcards, regex groups, and other bits.
+
+## Simple Route Config
+
+Let's look at the example from the routing page more closely. We'll flesh out
+the handler so that we end up with something like the following:
+
+```ts
+import { HandlerContext, RouteConfig } from "$fresh/server.ts";
+
+export const handler = {
+  GET(_req: Request, { params }: HandlerContext) {
+    console.log(params);
+    return new Response(params.path);
+  },
+};
+
+export const config: RouteConfig = {
+  routeOverride: "/x/:module@:version/:path*",
+};
+```
+
+Now if we hit the server with a request like
+`http://localhost:8000/x/bestModule@1.33.7/asdf`, then logging the params will
+show the following:
+
+```
+module: 'bestModule'
+version: '1.33.7'
+path: 'qqqqqqq'
+```
+
+## Complex Route Config
+
+Let's look at something a bit more complex. Issue 587 brings up the following
+example:
```

Comment:
Let's remove the "issue" part.

Reply:
Eagle eyed Marvin catches all references to issues 😅  Never again, don't worry!